ssoPlugin/control/settings/index.html

```diff
@@ -0,0 +1,194 @@
+
+		/**
+		 * Updates the alert message placeholder with the provided message and type.
+		 *
+		 * Expected types are "error" or "success". Default is "notification" and has no special style.
+		 *
+		 * @param {String}  message      The message that should be displayed.
+		 * @param {String} messageType? The message type, from the list above.
+		 */
+		const alertMessage = (message, messageType = 'notification') => {
+			const filteredClassNames = $apiSettingsAlertMessage.className
+				.split(' ')
+				.filter((className) => !className.startsWith('bif-sso__auth-api-settings__alert-message--'))
+			if (message) filteredClassNames.push(`bif-sso__auth-api-settings__alert-message--${messageType}`)
+			$apiSettingsAlertMessage.className = filteredClassNames.join(' ').trim()
+			$apiSettingsAlertMessage.innerHTML = message
+		}
+
+		/**
+		 * Validates that the required settings exist and aren't falsey.
+		 *
+		 * @param {Object} settings An object containing the settings.
+		 */
+		const areValidSettings = (settings) => {
+			return settings && settings.domain && settings.clientId
+		}
+
+		/**
+		 * Prefills the settings form with the provided values, if they're set.
+		 *
+		 * @param {Object} settings An object containing the settings.
+		 */
+		const prefillSettingsForm = (settings) => {
+			$apiSettingsFieldDomain.value = settings.domain || ''
+			$apiSettingsFieldClientId.value = settings.clientId || ''
+		}
+
+		/**
+		 * Saves the settings to the remote BuildFire datastore.
+		 *
+		 * TODO: Make this a Promise, probably.
+		 *
+		 * @param {Object} settings An object containing the settings.
+		 */
+		const saveSettings = (settings) => {
+			toggleForm($apiSettingsForm)
+			buildfire.datastore.save(settings, 'bif__sso__auth_api_settings', (err, status) => {
+				toggleForm($apiSettingsForm)
+				if (err || !status) alertMessage('❌ Unable to save settings', 'error')
+				else alertMessage('✅ Settings saved!', 'success')
+			})
+		}
+
+		/**
+		 * Handler for the form submission that saves to the BuildFire datastore.
+		 *
+		 * @param {SubmitEvent} e The native submit event fired by the browser.
+		 */
+		const handleSettingsFormSubmit = (e) => {
+			// Stop the native form submission
+			e.preventDefault()
+
+			// Clear the alert message
+			alertMessage('')
+
+			const settings = {
+				domain: $apiSettingsFieldDomain.value,
+				clientId: $apiSettingsFieldClientId.value,
+			}
+
+			if (areValidSettings(settings)) saveSettings(settings)
+			else alertMessage('❌ Please provide a valid domain and client ID.', 'error')
+		}
+
+		// ============================================================================================
+		// INIT/START
+		// ============================================================================================
+
+		ready(() => {
+			$apiSettingsForm.addEventListener('submit', handleSettingsFormSubmit)
+
+			// Fetch and prefill the existing auth API settings.
```

Comment:
In the spirit of "comments indicate the need for a helper method" it might be good to put this in a `loadAuthApiSettings` method.

Reply:
Sure!